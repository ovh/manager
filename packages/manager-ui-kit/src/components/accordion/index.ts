export { Accordion } from './Accordion.component';
export type { AccordionProps } from './Accordion.props';

export { AccordionContent } from './accordion-content/AccordionContent.component';
export { AccordionItem } from './accordion-item/AccordionItem.component';
export { AccordionTrigger } from './accordion-trigger/AccordionTrigger.component';
