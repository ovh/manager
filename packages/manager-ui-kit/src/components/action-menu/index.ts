export { ActionMenu } from './ActionMenu.component';
export type { ActionMenuItem, ActionMenuProps } from './ActionMenu.props';
