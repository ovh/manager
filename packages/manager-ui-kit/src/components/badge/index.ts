export { Badge } from './badge.component';

export type { BadgeProps } from './badge.props';

export * from './Badge.constants';
