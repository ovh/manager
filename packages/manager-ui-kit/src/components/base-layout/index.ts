export { BaseLayout } from './BaseLayout.component';

export type { BaseLayoutProps } from './BaseLayout.props';

export type { HeaderProps } from './header/Header.props';
