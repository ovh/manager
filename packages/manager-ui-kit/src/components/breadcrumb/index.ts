export { Breadcrumb } from './Breadcrumb.component';
export type { BreadcrumbProps } from './Breadcrumb.props';
export * from './Breadcrumb.constants';
