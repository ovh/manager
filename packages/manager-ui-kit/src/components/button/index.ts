export { Button } from './Button.component';
export type { ButtonProps } from './Button.props';
