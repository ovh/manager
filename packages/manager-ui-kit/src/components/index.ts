export * from './action-banner';
export * from './action-menu';
export * from './redirection-guard';
export * from './breadcrumb';
export * from './base-layout';
export * from './clipboard';
export * from './step';
export * from './tabs';
export * from './tiles-input-group';
export * from './tiles-input';

export * from './Link';
export * from './drawer';
export * from './guide-menu';

export * from './notifications';

export * from './price';

export * from './filters';

export * from './button';
export * from './text';

export * from './service-state-badge';

export * from './tile';
export * from './order';

export * from './badge';
export * from './modal';
export * from './tags-list';
export * from './tags-tile';
export * from './changelog-menu';
export * from './link-card';
export * from './error';
export * from './error-boundary';
export * from './update-name-modal';
export * from './delete-modal';
export * from './grid-layout';
export * from './onboarding-layout';
export * from './datagrid';
export * from './accordion';
export * from './checkbox';
export * from './combobox';
export * from './datepicker';
export * from './file-upload';
export * from './form-field';
export * from './message';
export * from './phone-number';
export * from './popover';
export * from './progress';
export * from './quantity';
export * from './radio-group';
export * from './select';
export * from './switch';
export * from './timepicker';
export * from './toggle';
export * from './tooltip';
export * from './tree-view';
