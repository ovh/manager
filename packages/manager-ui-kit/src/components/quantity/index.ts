export { Quantity } from './Quantity.component';
export type { QuantityProps } from './Quantity.props';

export { QuantityControl } from './quantity-control/QuantityControl.component';
export { QuantityInput } from './quantity-input/QuantityInput.component';
