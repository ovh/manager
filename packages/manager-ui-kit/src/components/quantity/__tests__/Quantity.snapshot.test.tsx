import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Quantity, QuantityControl, QuantityInput } from '@/components';

describe('Quantity Snapshot tests', () => {
  it('renders the component with default props and children', () => {
    const { container } = render(
      <Quantity max={10}>
        <QuantityControl>
          <QuantityInput />
        </QuantityControl>
      </Quantity>,
    );
    expect(container).toMatchSnapshot();
  });
});
