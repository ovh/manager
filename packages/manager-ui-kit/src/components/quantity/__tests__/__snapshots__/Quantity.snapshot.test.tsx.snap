// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Quantity Snapshot tests > renders the component with default props and children 1`] = `
<div>
  <div
    data-ods="quantity"
    data-part="root"
    data-scope="number-input"
    dir="ltr"
    id="number-input::r0:"
  >
    <div
      aria-disabled="false"
      class="_quantity-control_7aeg8_2"
      data-ods="quantity-control"
      data-part="control"
      data-scope="number-input"
      dir="ltr"
      role="group"
    >
      <button
        aria-hidden="true"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--sm_1koyw_34 _button--outline_1koyw_259 _quantity-control__decrement_7aeg8_8"
        data-ods="button"
        tabindex="-1"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--minus_10c23_376"
          data-ods="icon"
          role="presentation"
        />
      </button>
      <input
        aria-roledescription="numberfield"
        aria-valuemax="10"
        aria-valuemin="-9007199254740991"
        autocomplete="off"
        autocorrect="off"
        class="_quantity-input_cpyn3_2"
        data-ods="quantity-input"
        data-part="input"
        data-scope="number-input"
        dir="ltr"
        id="number-input::r0::input"
        inputmode="decimal"
        pattern="-?[0-9]*(.[0-9]+)?"
        role="spinbutton"
        spellcheck="false"
        type="text"
        value=""
      />
      <button
        aria-hidden="true"
        class="_button_1koyw_2 _button--primary_1koyw_100 _button--sm_1koyw_34 _button--outline_1koyw_259 _quantity-control__increment_7aeg8_15"
        data-ods="button"
        tabindex="-1"
        type="button"
      >
        <span
          class="_icon_10c23_2 _icon--plus_10c23_409"
          data-ods="icon"
          role="presentation"
        />
      </button>
    </div>
  </div>
</div>
`;
